### PR TITLE
feat(api): add api spec for processors resource

### LINKED ISSUE #185: Add all CRUD API's for the custom javascript resource to openapi3 definition. resource name "processors"

### PR DESCRIPTION
**Pull request by enudler**
fix #185

### CODE REVIEW

File: docs/openapi3.yaml [hunk 1]
```diff
@@ -19,39 +19,43 @@ info:
...
-      This resource allows you to configure Predator programmatically.
+- name: DSL Definitions
+  description: |
+    Predator's Domain Specific Language (DSL) allows you to generate request templates, which you can then reuse in the same test and in other tests, reducing replication.
+- name: Tests
+  description: |
+    Tests include end-to-end scenarios that are executed at pre-configured intervals to provide in-depth performance metrics of your API.
+- name: Processors
+  description: |
+    Processor file allows Preadtor execute custom javascript code.
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: Processor files allow Predator to execute custom javascript code during test flows.

File: docs/openapi3.yaml [hunk 2]
```diff
@@ -1010,6 +1016,194 @@ paths:
...
+      summary: Retrieve all processors files
+      description: Retrieve the details of all processor files.
+      parameters:
+      - in: query
+        name: from
+        description: From which result to start retrieve data ( default is zero )
+        required: false
+        schema:
+          type: integer
+          minimum: 0
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: add `default: 0`

File: docs/openapi3.yaml [hunk 3]
```diff
@@ -1010,6 +1016,194 @@ paths:
...
+    get:
+      operationId: retrieve-all-processor-files
+      tags:
+      - Processors
+      summary: Retrieve all processors files
+      description: Retrieve the details of all processor files.
+      parameters:
+      - in: query
+        name: from
+        description: From which result to start retrieve data ( default is zero )
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: remove ( default is zero )

File: docs/openapi3.yaml [hunk 4]
```diff
@@ -1010,6 +1016,194 @@ paths:
...
+        required: true
+    get:
+      operationId: retrieve-all-processor-files
+      tags:
+      - Processors
+      summary: Retrieve all processors files
+      description: Retrieve the details of all processor files.
+      parameters:
+      - in: query
+        name: from
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: are these needed? i feel like the amount of `processor files` created is limited and not sure how useful the query params are

File: docs/openapi3.yaml [hunk 5]
```diff
@@ -1010,6 +1016,194 @@ paths:
...
+        schema:
+          type: integer
+          minimum: 0
+      - in: query
+        name: limit
+        description: Max results to return from this query ( Limit to 100 )
+        required: false
+        schema:
+          type: integer
+          maximum: 100
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: add `default: 50`

File: docs/openapi3.yaml [hunk 6]
```diff
@@ -1479,30 +1673,69 @@ components:
...
+          description: The date and time that the procesor file was updated.
+          readOnly: true
+        created_at:
+          type: string
+          format: date-time
+          description: The date and time that the procesor file was created.
+          readOnly: true
+        javascript:
+          type: string
+          description: Javascript raw that will be used by the processor resource.
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: change to 'Raw javascript that will be used by the processor resource.'

File: docs/openapi3.yaml [hunk 7]
```diff
@@ -1723,3 +1956,32 @@ components:
...
+      description: Download processor file from the file_url and store it.
+      required:
+      - file_url
+      properties:
+        file_url:
+          type: string
+          description: url that the file can be downloaded from, should be direct download link.
+        force_download:
+          type: boolean
+          description: wether to try download the file_url on every invoke by a test.
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: whether

File: docs/openapi3.yaml [hunk 8]
```diff
@@ -1479,30 +1673,69 @@ components:
...
+      - properties:
+          run_id:
+            type: string
+            description: The run id. Only shown if the job is ran immediately.
+    processor:
+      discriminator:
+        propertyName: type
+      oneOf:
+      - $ref: '#/components/schemas/download'
+      - $ref: '#/components/schemas/raw'
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: i suggest renaming to `raw_javascript`

File: docs/openapi3.yaml [hunk 9]
```diff
@@ -1479,30 +1673,69 @@ components:
...
+      - $ref: '#/components/schemas/job_response'
+      - properties:
+          run_id:
+            type: string
+            description: The run id. Only shown if the job is ran immediately.
+    processor:
+      discriminator:
+        propertyName: type
+      oneOf:
+      - $ref: '#/components/schemas/download'
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: i suggest renaming to `file_download`

File: docs/openapi3.yaml [hunk 10]
```diff
@@ -1723,3 +1956,32 @@ components:
...
+        javascript:
+          type: string
+          description: Javascript as a string.
+    processor_type:
+      type: string
+      description: |
+        The type of the processor resource. Can be one of the following:
+        * `raw`: Raw javascript string that will be stored and persisted as it is.
+        * `download`: Predator will download the custom javascript by file_url and persist it.
+        enum:
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: remove the `enum: dsl, basic`

File: docs/openapi3.yaml [hunk 11]
```diff
@@ -1723,3 +1956,32 @@ components:
...
               type: number
+    download:
+      description: Download processor file from the file_url and store it.
+      required:
+      - file_url
+      properties:
+        file_url:
+          type: string
+          description: url that the file can be downloaded from, should be direct download link.
+        force_download:
```
Comment (NivLipetz) [inline docs/openapi3.yaml]: move this parameter to be under `Job` resource. If job has a processor_file, then user has the ability to force download it or not. (Default: false)

**Comment by enudler** [comment]
@NivLipetz all comments addressed

**Pull request merged**